### PR TITLE
Add ContainerPostProcessor for PanelContainer changes

### LINKED ISSUE #2240: [Carousel][MSM] Child items order get lost when inheritance get canceled
Steps to reproduce:
Create a page with a carousel component and inside three image components
insert three different images
rollout the page
open livecopy page 
cancel inheritance of carousel
open edit dialog of carousel and delete the second image component; close dialog.
=> the deleted image component is transfered to a ghost component, showing the frame "Placeholder for a deleted inherited Component"

Expected Behavior:
The order of the image components should not change.
Clicking on the frame "Placeholder for a deleted inherited Component" should show the symbol to revert the inheritance.

Experienced Behavior:
The order of the image components is lost - the deleted image is moved to the first position, but should remain at original place (second).
Clicking on the frame "Placeholder for a deleted inherited Component" does not show the symbol to revert the inheritance.

### LINKED ISSUE #2314: [Accordion] Deleting items resets all changes to dialog
## Bug Report

**Current Behavior**
When an item is deleted from the Accordion dialog (and likely other panel containers), the entire dialog is reset to the state of the JCR.
This means that all other edits that have been made in the dialog are undone. 
This also happens if the user cancels the delete when prompted with the "You are going to delete the selected component(s)" warning.
If multiple panels have been added, they are all removed. 

A secondary, but likely related, issue is that if an author deletes an existing item, but then subsequently cancels the entire edit dialog, the removal of the item is made in the JCR but the component does not refresh. This leads to a mismatch where the author can continue to interact with the deleted panel in the authoring interface until page refresh, but the underlying JCR nodes are already gone. 

**Expected behavior/code**

1. Deleting a panel from the Accordion dialog should not reset the rest of the dialog
2. Canceling the deletion of a panel from the Accordion dialog should have no effect on the dialog at all
3. If an author has deleted a panel but subsequently cancels the entire edit dialog there should be no changes to the component or it's underlying JCR nodes - Authors expect that canceling an edit dialog cancels any edits they may have made. 

**Environment**
- AEM 6.5.14
- Core Components 2.21.0
- JRE version 11

**Possible Solution**
Not sure, but looking in the `childreneditor.js` might be a place to start.

**Additional context / Screenshots**
----

**Reset on panel deletion**
This video shows the entire Accordion dialog being reset when a panel is deleted.

https://user-images.githubusercontent.com/19987287/190663892-527a79e8-96b5-4c3b-ae2d-fde324d741a1.mp4

----

**Reset on panel deletion cancelled**
This video shows the entire Accordion dialog being reset when a panel deletion is cancelled.

https://user-images.githubusercontent.com/19987287/190664216-14325f7e-a95d-4e32-b945-11f0bcedaebe.mp4

----

**Panel deleted but dialog cancelled**
This video shows the mismatch that occurs when a panel is deleted but then the dialog is cancelled.

https://user-images.githubusercontent.com/19987287/190664660-2d115b17-cb10-4f5b-a993-111906a0b9dc.mp4

### PR DESCRIPTION
**Pull request by bpauli**
- add ContainerPostProcessor to handle changes in a PanelContainer component like Accordion
- remove dedicated ContainerServlet
- revert confirmation dialog inside the children editor

fixes #2314 and #2240

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #2314, Fixes #2240  
| Patch: Bug Fix?          | 👍
| Minor: New Feature?      | 👎
| Major: Breaking Change?  | 👎
| Tests Added + Pass?      | Yes
| Documentation Provided   | Yes (code comments and or markdown)
| Any Dependency Changes?  |
| License                  | Apache License, Version 2.0

### CODE REVIEW

File: .github/codecov.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+coverage:
```
Comment (vladbailescu) [inline .github/codecov.yml:1]: Could we move this to a separate PR?

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/ContainerPostProcessor.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    protected void handleOrder(Resource container, SlingHttpServletRequest request) throws RepositoryException {
+        String[] orderedChildrenNames = StringUtils.split(request.getParameter(PARAM_ORDERED_CHILDREN), ",");
+        if (orderedChildrenNames != null && orderedChildrenNames.length > 0) {
+            final Node containerNode = container.adaptTo(Node.class);
+            if (containerNode != null) {
+                for (int i = orderedChildrenNames.length - 1; i >= 0; i--) {
+                    if (i == orderedChildrenNames.length - 1 && containerNode.hasNode(orderedChildrenNames[i])) {
+                        containerNode.orderBefore(orderedChildrenNames[i], null);
+                    } else if (containerNode.hasNode(orderedChildrenNames[i]) && containerNode.hasNode(orderedChildrenNames[i])) {
```
Comment (vladbailescu) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/ContainerPostProcessor.java:71]: There's something fishy about the logic here, the condition `containerNode.hasNode(orderedChildrenNames[i]) && containerNode.hasNode(orderedChildrenNames[i])` looks wrong, I'm guessing the second index should be `[i+1]`. I would also restructure the `if` clauses to eliminate the common part:
```
if (containerNode.hasNode(orderedChildrenNames[i])) {
    if (i == orderedChildrenNames.length - 1) {
        ...
    } else if (containerNode.hasNode(orderedChildrenNames[i+1]) {
        ...
    }
}

File: bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/services/ContainerPostProcessorTest.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    private final AemContext context = new AemContext(ResourceResolverType.JCR_MOCK);
+    private static final String PARAM_ORDERED_CHILDREN = "itemOrder";
+    private static final String PARAM_DELETED_CHILDREN = "deletedItems";
+
+    @Test
+    public void testOrderChildren() throws IOException, RepositoryException {
+        context.create().resource("/test/dummy/container");
+        context.create().resource("/test/dummy/container/child1");
+        context.create().resource("/test/dummy/container/child2");
+        String[] reorderedChildren = new String[]{"child2","child1"};
```
Comment (Buuhuu) [inline bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/services/ContainerPostProcessorTest.java:57]: Would it make sense to add a non-existing child here? This would cover the checks if the child exists.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/ContainerPostProcessor.java (hunk 2 above)
Comment (bpauli) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/ContainerPostProcessor.java:71]: This logic was already in place in the old ContainerServlet https://github.com/adobe/aem-core-wcm-components/blob/main/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/ContainerServlet.java#L88

Comment (bpauli) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/ContainerPostProcessor.java]: updated logic in 8d44260f19687f5ebf8bdcce15351ac950aab624

File: bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/services/ContainerPostProcessorTest.java (hunk 3 above)
Comment (bpauli) [reply bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/services/ContainerPostProcessorTest.java]: extended unit test in 8d44260f19687f5ebf8bdcce15351ac950aab624

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/ContainerPostProcessor.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.adobe.cq.wcm.core.components.internal.models.v1.PanelContainerImpl;
+
+@Component(service = SlingPostProcessor.class,
+           property = {
+                   Constants.SERVICE_RANKING + ":Integer=" + Integer.MAX_VALUE
+           })
+public class ContainerPostProcessor implements SlingPostProcessor {
```
Comment (jckautzmann) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/ContainerPostProcessor.java:41]: Please add some javadocs here and also to the below methods.
